src/implementation/c/index.ts

```diff
@@ -49,6 +49,11 @@ export class CCompiler {
     out.push('#endif  /* __SSE4_2__ */');
     out.push('');
 
+    out.push('#ifdef __ARM_NEON__');
```

Comment:
If this is limited to `arm64` as PR title states it should be followed by `#ifdef __aarch64__` otherwise 32-bit ARM targets with NEON are also included.